clifford/_conformal_layout.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+
+from ._layout import Layout
+from ._multivector import MultiVector
+
+
+class ConformalLayout(Layout):
+    r"""
+    A layout for a conformal algebra, which adds extra constants and helpers.
+
+    Typically these should be constructed via :func:`clifford.conformalize`.
+
+    Attributes
+    ----------
+    ep : MultiVector
+        The first added basis element, $e_{+}$, usually with $e_{+}^2 = +1$
+    en : MultiVector
+        The first added basis element, $e_{-}$, usually with $e_{-}^2 = -1$
+    eo : MultiVector
+        The null basis vector at the origin, $e_o = 0.5(e_{-} - e_{+})$
+    einf : MultiVector
+        The null vector at infinity, $e_\infty = e_{-} + e_{+}$
+    E0 : MultiVector
+        The minkowski subspace bivector, $e_\infty ^ e_o$
+    I_base : MultiVector
+        The pseudoscalar of the base ga, in cga layout
+    """
+    def __init__(self, *args, layout=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._base_layout = layout
+
+        basis_vectors = self.basis_vectors
+        added_keys = sorted(basis_vectors.keys())[-2:]
+        ep, en = [basis_vectors[k] for k in added_keys]
+
+        # setup  null basis, and minkowski subspace bivector
+        eo = .5 ^ (en - ep)
+        einf = en + ep
+        E0 = einf ^ eo
```

Comment:
Note that this is just copied from the old code - I haven't changed any definitions here

Reply:
i like this change.